src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -1001,22 +1010,24 @@ impl Deref for GraphEditorModelWithNetwork {
     }
 }
 
+/// Context data required to create a new node.
+#[derive(Debug)]
+struct NodeContext<'a> {
+    pointer_style  : &'a frp::Source<cursor::Style>,
+    tooltip_update : &'a frp::Source<tooltip::Style>,
+    output_press   : &'a frp::Source<EdgeEndpoint>,
+    input_press    : &'a frp::Source<EdgeEndpoint>,
+    output         : &'a FrpEndpoints,
+}
```

Comment:
1. Very cool.
2. I'd name it `NewNodeData` or something likethat to indicate its purpose